nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-parameter-contexts.js

```diff
@@ -0,0 +1,2092 @@
+            var parameter = parameterData.getItem(args.row);
+
+            if (parametersGrid.getColumns()[args.cell].id === 'actions') {
+                var target = $(e.target);
+
+                // determine the desired action
+                if (target.hasClass('delete-parameter')) {
+                    // mark the property in question for removal and refresh the table
+                    parameterData.updateItem(parameter.id, $.extend(parameter, {
+                        hidden: true
+                    }));
+
+                    // reset the selection if necessary
+                    var selectedRows = parametersGrid.getSelectedRows();
+                    if (selectedRows.length === 0) {
+                        parametersGrid.setSelectedRows([0]);
+                    }
+
+                    var rows = parameterData.getItems();
+
+                    if (rows.length === 0) {
+                        // clear usages
+                        resetUsage();
+                    } else {
+                        var reset = true;
+                        $.each(rows, function (_, parameter) {
+                            if (!parameter.hidden) {
+                                reset = false;
+                            }
+                        });
+
+                        if (reset) {
+                            // clear usages
+                            resetUsage();
+                        }
+                    }
+
+                    // update the buttons to possibly trigger the disabled state
+                    $('#parameter-context-dialog').modal('refreshButtons');
+
+                    // prevents standard edit logic
+                    e.stopImmediatePropagation();
+                } else if (target.hasClass('edit-parameter')) {
+                    $('#parameter-dialog').modal('setHeaderText', 'Edit Parameter').modal('setButtonModel', [{
+                        buttonText: 'Apply',
+                        color: {
+                            base: '#728E9B',
+                            hover: '#004849',
+                            text: '#ffffff'
+                        },
+                        disabled: function () {
+                            var value = $('#parameter-value-field').val();
+                            var description = $('#parameter-description-field').val();
+
+                            if (value === parameter.value) {
+                                if (value === '********') {
+                                    return false;
+                                } else if (value === '' && $('#parameter-dialog').find('.nf-checkbox').hasClass('checkbox-unchecked')) {
+                                    return true;
+                                } else if ($('#parameter-dialog').find('.nf-checkbox').hasClass('checkbox-checked')) {
+                                    return false;
+                                }
+                
```

Comment:
The `color` definitions for `Apply` and `Cancel` buttons are duplicated a few times. Maybe extract them into variables that could be re-used? This would keep color changes in the future to a single location.

Reply:
Agreed. However I think this is more of a general issue throughout NiFi and we can open an issue for it.